client_integration_test.go

```diff
@@ -873,6 +873,44 @@ func TestClientChmodReadonly(t *testing.T) {
 	}
 }
 
+func TestClientSetuid(t *testing.T) {
+	skipIfWindows(t) // No UNIX permissions.
+	if *testServerImpl {
+		t.Skipf("skipping with -testserver")
+	}
+
+	sftp, cmd := testClient(t, READWRITE, NODELAY)
+	defer cmd.Wait()
+	defer sftp.Close()
+
+	f, err := ioutil.TempFile("", "sftptest-setuid")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(f.Name())
+	f.Close()
+
+	const perm = 0700 | os.ModeSetuid
+
+	err = sftp.Chmod(f.Name(), perm)
+	require.NoError(t, err)
+
+	info, err := sftp.Stat(f.Name())
+	require.NoError(t, err)
+	require.Equal(t, perm, info.Mode()&perm)
```

Comment:
This test is insufficient. `info.Mode()&perm` would not catch spare additional bits in `info.Mode()` So if our implementation were bugged and applied `04755` to the file, while this should fail, here it would not.

Reply:
Fixed.